libraries/common/bytearray.effekt

```diff
@@ -133,6 +133,40 @@ extern def toString(arr: ByteArray) at {}: String =
   chez "(utf8->string ${arr})"
   vm "bytearray::toString(ByteArray)"
 
+def toHexCharBytes(b: Byte): Unit / emit[Byte] = {
+  val unsigned: Int = b.toInt.bitwiseAnd(0xFF.toInt)
+
+  val high = unsigned.bitwiseShr(4).bitwiseAnd(0xFF.toInt)
+  val low = unsigned.bitwiseAnd(0x0F.toInt)
+  println(b.toInt.show ++ " (U" ++ unsigned.show ++ ") := " ++ high.show ++ "; " ++ low.show)
+
+  def toHexDigit(n: Int): Byte =
+    if (n < 10) {
+      (n + 48)       // '0' = 48
+    } else {
+      (n - 10 + 65)  // 'A' = 65
+    }.toByte
+
+  do emit(high.toHexDigit)
+  do emit(low.toHexDigit)
+}
```

Comment:
The `show` instance for `Byte`s is a bit weird since we're trying to write it in Effekt directly.
In order to do _anything_ at all, we need to either convert to characters or use bytearrays -- I've chosen to do the latter here, but as a consequence, the `println` and `show` for `Byte` are defined awkwardly in `bytearray`...

`show(b: Byte) := bytearray::collect { b.toHexCharBytes }`